packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -150,6 +150,24 @@ abstract class _UbiquitousInheritedWidget extends InheritedWidget {
 ///
 /// For example, [Padding] depends on the [Directionality] to resolve
 /// [EdgeInsetsDirectional] objects into absolute [EdgeInsets] objects.
+///
+/// {@tool snippet}
+///
+/// This example uses a right-to-left text direction and draws a blue box with
+/// a right margin of 8 pixels.
+///
+/// ```dart
+/// Directionality(
+///   textDirection: TextDirection.rtl,
+///   child: Padding(
+///     padding: const EdgeInsetsDirectional.only(start: 8),
+///       child: Container(
+///       color: Colors.blue,
+///     ),
+///   ),
```

Comment:
Since the documentation describes "a right margin", here's an optional suggestion:

```suggestion
///   child: Container(
///     margin: const EdgeInsetsDirectional.only(start: 8),
///     color: Colors.blue,
///   ),
```

Reply:
Alternatively:

```suggestion
///   child: Padding(
///     padding: const EdgeInsetsDirectional.only(start: 8),
///     child: Container(
///       color: Colors.blue,
///     ),
///   ),
```

Reply:
Ah nice, thanks for the simplification!